rajawali/src/main/java/org/rajawali3d/primitives/Cylinder.java

```diff
@@ -0,0 +1,157 @@
+ *
+ */
+public class Cylinder extends Object3D {
+
+	private final float PI = (float) Math.PI;
+	private float mLength;
+	private float mRadius;
+	private int mSegmentsC;
+	private int mSegmentsL;
+	private boolean mCreateTextureCoords;
+	private boolean mCreateVertexColorBuffer;
+	private Sphere s;
+
+	/**
+	 * Creates a cylinder primitive. Calling this constructor will create texture coordinates but no vertex color buffer.
+	 *
+	 * @param length
+	 *			The length of the cylinder
+	 * @param radius
+	 *			The radius of the cylinder
+	 * @param segmentsL
+	 *			The number of length segments
+	 * @param segmentsC
+	 *			The number of circle segments
+	 */
+	public Cylinder(float length, float radius, int segmentsL, int segmentsC) {
+		this(length, radius, segmentsL, segmentsC, true, false, true);
+	}
+
+	/**
+	 * Creates a cylinder primitive.
+	 *
+	 * @param length
+	 *			The length of the cylinder
+	 * @param radius
+	 *			The radius of the cylinder
+	 * @param segmentsL
+	 *			The number of length segments
+	 * @param segmentsC
+	 *			The number of circle segments
+	 * @param createTextureCoordinates
+	 *			A boolean that indicates whether the texture coordinates should be calculated or not.
+	 * @param createVertexColorBuffer
+	 *			A boolean that indicates whether a vertex color buffer should be created or not.
+	 * @param createVBOs
+	 *			A boolean that indicates whether the VBOs should be created immediately.
+	 */
+	public Cylinder(float length, float radius, int segmentsL, int segmentsC, boolean createTextureCoordinates,
+					boolean createVertexColorBuffer, boolean createVBOs) {
+		super();
+		mLength = length;
+		mRadius = radius;
+		mSegmentsL = segmentsL;
+		mSegmentsC = segmentsC;
+		mCreateTextureCoords = createTextureCoordinates;
+		mCreateVertexColorBuffer = createVertexColorBuffer;
+		init(createVBOs);
+	}
+
+	protected void init(boolean createVBOs) {
+		int numVertices = (mSegmentsC + 1) * (mSegmentsL + 1);
+		int numIndices = 2 * mSegmentsC * mSegmentsL * 3;
+
+		float[] vertices = new float[numVertices * 3];
+		float[] normals = new float[numVertices * 3];
+		int[] indices = new int[numIndices];
+
+		int i, j;
+		int vertIndex = 0, index = 0;
+		final float normLen = 1.0f / mRadius;
+
+		for (j = 0; j <= mSegmentsL; ++j) {
+			float z = mLength * ((float)j/(float)mSegmentsL) - mLength / 2.0f;
+
+			for (i = 0; i <= mSegmentsC; ++i) {
+				float verAngle = 2.0f * PI * i / mSegmentsC;
+				float x = mRadius * (float) Math.cos(verAngle);
+				float y = mRadius * (float) Math.sin(verAngle);
+
+				normals[vertIndex] = x * normLen;
+				vertices[vertIndex++] = x;
+				normals[vertIndex] = y * normLen;
+				vertices[vertIndex++] = y;
+				normals[vertIndex] = 0;
+				vertices[vertIndex++] = z;
+				System.out.println("normal["+((vertIndex-3)/3) +"]="+normals[vertIndex-3]+","+normals[vertIndex-2]+","+normals[vertIndex-1]);/////
```

Comment:
System calls in general but especially System.out should be avoided on Android. In any event, this logging is verbose and should be removed/commented out.